WalletWasabi.Fluent/ViewModels/Wallets/Home/Tiles/PrivacyRing/PrivacyBarItemViewModel.cs

```diff
@@ -1,28 +1,53 @@
 using Avalonia;
 using Avalonia.Media;
+using System.Linq;
 using WalletWasabi.Blockchain.TransactionOutputs;
 using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.Models;
 using WalletWasabi.Fluent.ViewModels.Wallets.Advanced.WalletCoins;
+using WalletWasabi.Wallets;
 
 namespace WalletWasabi.Fluent.ViewModels.Wallets.Home.Tiles.PrivacyRing;
 
-public class PrivacyBarItemViewModel : WalletCoinViewModel
+public class PrivacyBarItemViewModel : ViewModelBase, IDisposable
 {
-	public PrivacyBarItemViewModel(PrivacyBarViewModel parent, SmartCoin coin, double start, double width) : base(coin)
+	public const double BarHeight = 10;
+
+	public PrivacyBarItemViewModel(PrivacyBarViewModel parent, SmartCoin coin, double start, double width)
 	{
+		Coin = new WalletCoinViewModel(coin);
+
 		IsPrivate = coin.IsPrivate(parent.Wallet.KeyManager.AnonScoreTarget);
 		IsSemiPrivate = !IsPrivate && coin.IsSemiPrivate();
 		IsNonPrivate = !IsPrivate && !IsSemiPrivate;
 
 		Data = new RectangleGeometry
 		{
-			Rect = new Rect((double)start, 0, (double)width, 10)
+			Rect = new Rect(start, 0, width, BarHeight)
+		};
+	}
+
+	public PrivacyBarItemViewModel(Pocket pocket, Wallet wallet, double start, double width)
+	{
+		IsPrivate = pocket.Coins.First().IsPrivate(wallet.KeyManager.AnonScoreTarget);
+		IsSemiPrivate = !IsPrivate && pocket.Coins.First().IsSemiPrivate();
```

Comment:
`.All()` instead of `First()` looks more secure to me. `Pocket.Empty` is an existing term where this line would throw an exception, and also it is more sure if the condition is true on all items instead of the first one. (Mixed pockets in the future maybe)